soa/service/s3.cc

```diff
@@ -401,7 +401,27 @@ performSync() const
 
         myRequest.get_info(CURLINFO_RESPONSE_CODE, responseCode);
 
-        if (responseCode >= 300 && responseCode != 404) {
+        /* Detect so-called "REST error", which may or may not have an HTTP
+           status >= 200
+           (http://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html)
```

Comment:
There is here: http://docs.aws.amazon.com/AmazonS3/latest/API/mpUploadComplete.html I don't know whether this is specific to CompleteMultipartUpload. I suggest we restrict this handling to non-GET requests.